### ISSUE TITLE
Factor/categorical color symbology in plots

### ISSUE THREAD

**Issue by RobertSellers** [Swarchal/platetools#27]
Are there any methods for allowing a set # of categorical colors for ggplots?  Everything defaults as a continuous color range.

**Comment by Swarchal** [comment]
Yeah it's not documented very well, but you can pass strings or factors to `raw_map`.

e.g:
```R
df = data.frame(
    well = num_to_well(1:96),
    sample_name = sample(c("a", "b", "c"), 96, replace=TRUE)
)

raw_map(
    well = df$well,
    data = df$sample_name,
    plate = 96
)
```

![plot](https://user-images.githubusercontent.com/10051679/106797130-2345e000-6654-11eb-8d0c-22d4e022a69d.png)

**Comment by RobertSellers** [comment]
Ah - no this should have been obvious to me.  I had some custom symbology running through it which was the error...

`+ scale_fill_viridis()`

**Comment by RobertSellers** [comment]
Hi again.  I just wanted to follow-up with something related.  Is it possible to add these discrete labels to the well fills?  Using your example - placing an "a" over all of the red wells.  Is this possible with platetools or does this require some very custom work?

**Comment by Swarchal** [comment]
Good question, and another undocumented workflow. You can add labels using `geom_text()` from ggplot.

```R
library(platetools)
library(ggplot2)

df = data.frame(
    well = num_to_well(1:96),
    sample_name = sample(c("a", "b", "c"), 96, replace=TRUE)
)

raw_map(
    well = df$well,
    data = df$sample_name,
    plate = 96
) + 
    geom_text(aes(label=df$sample_name))
```

![plot2](https://user-images.githubusercontent.com/10051679/106913754-3e6c2a80-66fc-11eb-8faf-db9b4e9fb533.png)

**Comment by RobertSellers** [comment]
Excellent, thank you.  This is looking quite good.
